src/components/shared/IdleSearchField.js

```diff
@@ -71,6 +69,9 @@ class IdleSearchField extends PureComponent {
   }
 
   _onClearButtonClick() {
+    clearTimeout(this._timeout);
+    this._timeout = 0;
+
```

Comment:
For somebody quick enough (this is achievable by using the keyboard only) it was possible to clear the input but still have the last value set.